vms/platformvm/state/stakers.go

```diff
@@ -190,10 +195,19 @@ func (v *baseStakers) DeleteDelegator(staker *Staker) {
 	v.pruneValidator(staker.SubnetID, staker.NodeID)
 
 	validatorDiff := v.getOrCreateValidatorDiff(staker.SubnetID, staker.NodeID)
-	if validatorDiff.deletedDelegators == nil {
-		validatorDiff.deletedDelegators = make(map[ids.ID]*Staker)
+	found := false
+	if validatorDiff.addedDelegators != nil {
+		if _, found = validatorDiff.addedDelegators.Get(staker); found {
+			// delegator to be removed was just added. Wipe it up here
+			validatorDiff.addedDelegators.Delete(staker)
+		}
+	}
+	if !found {
+		if validatorDiff.deletedDelegators == nil {
+			validatorDiff.deletedDelegators = make(map[ids.ID]*Staker)
+		}
+		validatorDiff.deletedDelegators[staker.TxID] = staker
```

Comment:
Property tests allows inserting and immediately deleting a delegator. If we allow such a behaviour, we must introduce this special handling, to avoid updating validators set in the wrong way.